iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -2172,6 +2178,80 @@ static STRING_HANDLE buildClientId(const char* device_id, const char* module_id)
     }
 }
 
+static int buildConfigForUsernameStep2IfNeeded(PMQTTTRANSPORT_HANDLE_DATA transport_data)
```

Comment:
buildConfigForUsernameStep2IfNeeded  -> buildConfigForUsernameForModelId

Reply:
It's more than just ModelId.  We also put the API version on here, the productInfo (if app set, or alternatly our default "C-SDK-version-blah/OS-version", ModelId.

I'm fine with you not liking `buildConfigForUsernameStep2IfNeeded ` :).  Let me know if you have a better suggestion that captures the various stuff set here.